### PR TITLE
quote str value with special char in dev output

### PR DESCRIPTION
**Pull request by trim21**
# Summary

https://github.com/hynek/structlog/issues/647

# Pull Request Check List

- [ ] Do **not** open pull requests from your `main` branch – **use a separate branch**!
  - There's a ton of footguns waiting if you don't heed this warning. You can still go back to your project, create a branch from your main branch, push it, and open the pull request from the new branch.
  - This is not a pre-requisite for your pull request to be accepted, but **you have been warned**.
- [x] Added **tests** for changed code.
    - The CI fails with less than 100% coverage.
- [ ] **New APIs** are added to our typing tests in [`api.py`](https://github.com/hynek/structlog/blob/main/tests/typing/api.py).
- [ ] Updated **documentation** for changed code.
    - [ ] New functions/classes have to be added to `docs/api.rst` by hand.
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
      - The next version is the second number in the current release + 1. The first number represents the current year. So if the current version on PyPI is 23.1.0, the next version is gonna be 23.2.0. If the next version is the first in the new year, it'll be 24.1.0.
- [ ] Documentation in `.rst` and `.md` files is written using [**semantic newlines**](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) are documented in the [**changelog**](https://github.com/hynek/structlog/blob/main/CHANGELOG.md).
- [ ] Consider granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so maintainers can fix minor issues themselves without pestering you.

### CODE REVIEW

File: src/structlog/dev.py [hunk 1]
```diff
@@ -703,6 +702,8 @@ def _repr(self, val: Any) -> str:
             return repr(val)
 
         if isinstance(val, str):
+            if set(val) & {" ", "\t", "=", "\n"}:
```
Comment (trim21) [inline src/structlog/dev.py:705]: any other characters should be quoted？

Comment (trim21) [reply src/structlog/dev.py:705]: maybe also single quote and double quote？

**Comment by trim21** [comment]
any other characters should be quoted？

File: src/structlog/dev.py [hunk 2]
```diff
@@ -694,7 +693,7 @@ def add_meaningless_arg(arg: str) -> None:
             Column("logger_name", logger_name_formatter),
         ]
 
-    def _repr(self, val: Any) -> str:
+    def value_repr(self, val: Any) -> str:
```
Comment (hynek) [inline src/structlog/dev.py:696]: I don't think we need to make this method public?

Comment (trim21) [reply src/structlog/dev.py]: OK

**Comment by hynek** [comment]
I have fixed the CI for 3.13; please either rebase on main or click "Update branch".

**Comment by hynek** [comment]
never mind, you're done here;, thanks!

**Comment by trim21** [comment]
I think you can also click "Update branch" to merge master into this branch?

**Comment by hynek** [comment]
Yes and I did, but I thought you're still fixing the other thing and didn't want to fiddle with your branch.

**Pull request merged**